src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+            TokenTypes.LITERAL_WHILE,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.OBJBLOCK,
+            TokenTypes.SLIST,
+            TokenTypes.STATIC_INIT,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtils.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+
+        DetailAST leftBrace = null;
+        switch (ast.getType()) {
+            case TokenTypes.ANNOTATION_ARRAY_INIT:
+            case TokenTypes.ARRAY_INIT:
+                leftBrace = ast;
+                break;
+            case TokenTypes.INTERFACE_DEF:
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.ANNOTATION_DEF:
+            case TokenTypes.ENUM_DEF:
+            case TokenTypes.ENUM_CONSTANT_DEF:
+            case TokenTypes.LITERAL_NEW:
+                // These tokens are for the fine tuning. For example, you can enable this for
+                // classes, but not for enums.
+                final DetailAST objBlock = ast.findFirstToken(TokenTypes.OBJBLOCK);
+                if (objBlock != null) {
+                    leftBrace = objBlock.getFirstChild();
+                }
+                break;
+            case TokenTypes.OBJBLOCK:
+                // A general case of class/interface/enum/enum field/annotation/anonymous class.
+                leftBrace = ast.getFirstChild();
+                break;
+            case TokenTypes.LITERAL_ELSE:
+                // Special handling for the "else if" case.
+                final DetailAST candidate = ast.getFirstChild();
+                if (isRightCurlyChildOf(candidate)) {
+                    leftBrace = candidate;
+                }
+                break;
+            case TokenTypes.SLIST:
+                // The scope block is a SLIST with another SLIST as the parent.
+                if (ast.getParent().getType() == TokenTypes.SLIST) {
+                    leftBrace = ast;
+                }
+                break;
+            case TokenTypes.LITERAL_SWITCH:
+                // The Switch does not have SLIST unlike the others.
+                leftBrace = ast.findFirstToken(TokenTypes.LCURLY);
+                break;
+            case TokenTypes.CASE_GROUP:
+                // The CASE_GROUP is the parent of 'case', 'default' and SLIST parent.
+                leftBrace = ast.getLastChild().findFirstToken(TokenTypes.SLIST);
+                break;
+            case TokenTypes.LITERAL_CASE:
+            case TokenTypes.LITERAL_DEFAULT:
+                // The 'case' and 'default' are siblings for SLIST parent.
+                leftBrace = ast.getParent().getLastChild().findFirstToken(TokenTypes.SLIST);
+                break;
+            default:
+                // All the rest: for/while/if/STATIC_INIT,lambda,etc.
+                leftBrace = ast.findFirstToken(TokenTypes.SLIST);
+                break;
+        }
+
+        if (leftBrace != null) {
+            verify(leftBrace);
```

Comment:
we need to move all `log` methods to `visitToken` method to let read algorithm easily. Please do.
We need to read whole algorithm of processing as english text in `visitToken` method.

Reply:
Done.